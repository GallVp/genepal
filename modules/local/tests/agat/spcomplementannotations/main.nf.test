nextflow_process {

    name "Test Process AGAT_SPCOMPLEMENTANNOTATIONS"
    script "../../../../gallvp/agat/spcomplementannotations/main.nf"
    process "AGAT_SPCOMPLEMENTANNOTATIONS"

    tag "modules"
    tag "modules_nfcore"
    tag "agat"
    tag "agat/spcomplementannotations"

    // Test to show that agat_sp_complement_annotations does not fuse two genes into one
    //
    // Issue #153: <https://github.com/Plant-Food-Research-Open/genepal/issues/153>
    test("Should not fuse two genes into one") {

        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    [
                        file("$baseDir" + '/modules/local/tests/agat/spmergeannotations/testdata/file1.gff', checkIfExists: true), // 10 genes
                    ]
                ]
                input[1] = [
                    [],
                    [
                        file("$baseDir" + '/modules/local/tests/agat/spmergeannotations/testdata/file2.gff', checkIfExists: true) // 5 genes
                    ]
                ]
                input[2] = []
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()

            def featureLines = file(process.out.gff[0][1]).text.readLines().findAll { ! it.startsWith('#') }
            def geneLines = featureLines.collect { it.split('\t') }.findAll { it[2] == 'gene' }

            with(geneLines) {
                assert size() == 15 // 10 + 5.
            }
        }

    }

}
