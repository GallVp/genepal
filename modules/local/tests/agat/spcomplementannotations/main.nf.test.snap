{
    "Should not fuse two genes into one": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff:md5,1d42a1519c1faf86188f64b0504d289d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3b461a98eb18641f842397b7c626f59c"
                ],
                "gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff:md5,1d42a1519c1faf86188f64b0504d289d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3b461a98eb18641f842397b7c626f59c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-06T10:49:26.699831"
    }
}
